crates/ruff_cli/src/commands/run.rs

```diff
@@ -109,8 +112,9 @@ pub(crate) fn run(
                         .and_then(|package| *package);
 
                     let settings = resolver.resolve_all(path, pyproject_config);
-                    let package_root = package.unwrap_or_else(|| path.parent().unwrap_or(path));
-                    let cache = caches.as_ref().and_then(|caches| caches.get(&package_root));
+
+                    let cache_root = package.unwrap_or_else(|| path.parent().unwrap_or(path));
+                    let cache = caches.as_ref().and_then(|caches| caches.get(&cache_root));
```

Comment:
(maybe for another) I understand the need to remove the `unwrap` as it was causing panics, but it does mean something went wrong before this state (as the `caches` should be filled if we reach this point). Maybe we should emit a warning or something?